backend/score.py

```diff
@@ -108,111 +147,95 @@ async def extract_knowledge_graph_from_file(
     Returns:
           Nodes and Relations created in Neo4j databse for the pdf file
     """
-    
-    if file:
-        return await asyncio.to_thread(
-            extract_graph_from_file,
-            uri,
-            userName,
-            password,
-            model,
-            database,
-            file=file
-        )
-    elif source_url:
-        return await asyncio.to_thread(
-            extract_graph_from_file,
-            uri,
-            userName,
-            password,
-            model,
-            database,
-            source_url=source_url,
-            aws_access_key_id=aws_access_key_id,
-            aws_secret_access_key=aws_secret_access_key,
-            wiki_query=wiki_query,
-            max_sources=max_sources,
-        )
-    elif wiki_query:
-        return await asyncio.to_thread(
-            extract_graph_from_file,
-            uri,
-            userName,
-            password,
-            model,
-            database,
-            wiki_query=wiki_query
-        )
-    elif gcs_bucket_name:
-        return await asyncio.to_thread(
-            extract_graph_from_file,
-            uri,
-            userName,
-            password,
-            model,
-            database,
-            gcs_bucket_name = gcs_bucket_name,
-            gcs_bucket_folder = gcs_bucket_folder,
-            gcs_blob_filename = gcs_blob_filename
-        )
-             
-            
-    else:
-        return {"status": "Failed", "error": "No file found"}
+    try:
+        if source_type == 'local file':
+            return await asyncio.to_thread(
```

Comment:
can you explain to me, why we're using asyncio here but not with the other requests? is this a pre-requesite to do multi-threading / async away further down?

Reply:
We are accepting Asynchronous calls only for extraction and it's not threading.